bin/imsim.py

```diff
@@ -98,24 +100,24 @@ def main():
                               m5=[defaults.m5(bandpass)],
                               seeing=[seeing])
 
-    # Trim the input catalog to a single chip.
-    raICRS = phoSimObjectList['ra'].values
-    decICRS = phoSimObjectList['dec'].values
-    phoSimObjectList['chipName'] = chipNameFromRaDec(raICRS, decICRS,
-                                                     camera=camera,
-                                                     obs_metadata=obs,
-                                                     epoch=2000.0)
     # Now further sub-divide the source dataframe into stars and galaxies.
     if arguments.sensor is not None:
+        # Trim the input catalog to a single chip.
+        raICRS = phoSimObjectList['ra'].values
+        decICRS = phoSimObjectList['dec'].values
+        phoSimObjectList['chipName'] = chipNameFromRaDec(raICRS, decICRS,
+                                                         camera=camera,
+                                                         obs_metadata=obs,
+                                                         epoch=2000.0)
         starDataBase = \
-            phoSimObjectList.query("galSimType == 'pointSource' and magNorm < 50 and chipName=='%s'" % arguments.sensor)
+            phoSimObjectList.query("galSimType=='pointSource' and magNorm<50 and chipName=='%s'" % arguments.sensor)
```

Comment:
It is definitely needed for the Twinkles instance catalogs.  Those can have tens of sources with magnorms ranging from 50 to 1000.  I didn't have the patience to find out exactly at which value the mean counts was zero, so I just picked 50.  Unless there is a practical reason for not having this cut, I don't see why we shouldn't retain it.